ipatests/test_webui/test_cert.py

```diff
@@ -45,7 +92,404 @@ def test_read(self):
         Certs don't have standard mod, add and delete methods.
         """
         self.init_app()
-        self.navigate_to_entity(ENTITY)
+        self.navigate_to_entity(data_cert.ENTITY)
         rows = self.get_rows()
         self.navigate_to_row_record(rows[0])
         self.navigate_by_breadcrumb("Certificates")
+
+    @screenshot
+    def test_search_subject(self):
+        """
+        Try to search certificate by subject
+        """
+        self.init_app()
+        self.navigate_to_entity(data_cert.ENTITY)
+        self.select('select[name=search_option]', 'subject')
+        search_pkey(self, 'Certificate Authority')
+        rows = self.get_rows()
+        assert len(rows) != 0
+
+        # try to search non-existent subject
+        self.navigate_to_entity(data_cert.ENTITY)
+        self.select('select[name=search_option]', 'subject')
+        search_pkey(self, 'nonexistent')
+        rows = self.get_rows()
+        assert len(rows) == 0
+
+        # try to search subject with speacial char
+        self.navigate_to_entity(data_cert.ENTITY)
+        self.select('select[name=search_option]', 'subject')
+        search_pkey(self, '<,>.?/')
+        rows = self.get_rows()
+        assert len(rows) == 0
+
+        # try to search subject with leading space
+        check_space_error(self, ' Certificate Authority', 'subject')
+
+        # try to search subject with trailing space
+        check_space_error(self, 'Certificate Authority ', 'subject')
+
+    @screenshot
+    def test_search_revocation_reason(self):
+        """
+        Try to search certificates by revocation reason
+        """
+        self.init_app()
+        self.navigate_to_entity(data_cert.ENTITY)
+
+        # add a new cert.
+        self.facet_button_click('request_cert')
+        self.fill_textbox('principal', 'HTTP/master.testrelm.test')
+        self.check_option('add', 'checked')
+        self.fill_textarea('csr', data_cert.CERT_CSR)
+        self.dialog_button_click('issue')
```

Comment:
It would be good, for readability, to break the setup (the cert request and revocation) into a separate method, preceding the others.